### PR TITLE
 #64 chore: vitest-setup

### PR DESCRIPTION
**Pull request by patsicko**
# What this PR do
- Setup vitest for testing
# Description of the task 
- Install vitest and other related dependances
- Configure vite to use jsdom for testing
- Add test in package.json scripts to run test 
# How should this be manually tested?
- clone the repo using git clone repo
- change branch by running git checkout ch-vitest-setup
- install dependencies by npm install
- run npm run test passed tests and coverage will be displayed on CLI
test

### CODE REVIEW

**Comment by patsicko** [comment]
[Setup tests](https://trello.com/c/LIEwrHcP/61-setup-tests)

**Review by oliviertech212 — APPROVED** [review]

**Review by Shumba250 — APPROVED** [review]

**Review by Shumba250 — CHANGES_REQUESTED** [review]
the coverage always shows 100% even if the tests fail. try solving it

**Review by wakaflorien — APPROVED** [review]

File: src/tests/App.test.jsx
Comment (sostene12) [inline src/tests/App.test.jsx:1]: - even if the text is wrong , it still passes

**Review by sostene12 — CHANGES_REQUESTED** [review]

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- add vitest ui for better visual display
- add a separate test:run command without coverage 
```
   "test:ui": "vitest --ui",
    "test:run": "vitest run",
``` 
something like this.

**Review by wakaflorien — APPROVED** [review]

**Review by sostene12 — APPROVED** [review]

**Review by Shumba250 — APPROVED** [review]

**Pull request merged**